python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,841 @@
-                return create_enum_class(schema_definition)
-            elif schema_definition.get_root_key() != "schema":
+        else:
+            parent_classes = inheritance_parents  # the following causes a method resolution order (MRO) error when creating the type: [object] + inheritance_parents
+            try:
+                instance_class = type(
+                    schema_definition.get_python_class_name(),
+                    tuple(parent_classes),
+                    {"__module__": schema_definition.get_python_module_name()},
+                )
+            except LanguageError as e:
                 raise LanguageError(
-                    f"Definition {schema_definition.name} is not a schema",
-                    get_location_str(
-                        schema_definition.get_root_key(), schema_definition.lexemes
+                    f"Failed to create instance_class for {schema_definition.name}: {e.message}",
+                    self.get_location_str(
+                        schema_definition.name, schema_definition.lexemes
                     ),
                 )
+        return instance_class
+
+    def create_schema_class(self, schema_definition: Definition) -> Type:
+        """
+        Creates a schema class from a given schema definition.
+
+        Args:
+            schema_definition (Definition): A schema definition to convert to a class.
+
+        Returns:
+            The created class.
+        """
+        instance_class = None
+
+        fully_qualified_name = schema_definition.get_fully_qualified_name()
+        if schema_definition.get_root_key() == "primitive":
+            # this is a primitive, so there's no structure to create...just return the python type
+            return eval(self.primitive_name_to_py_typee[schema_definition.name])
+        elif schema_definition.get_root_key() == "enum":
+            # this is an enum, so create the enum class
+            return self.create_enum_class(schema_definition)
+        elif schema_definition.get_root_key() != "schema":
+            raise LanguageError(
+                f"Definition {schema_definition.name} is not a schema",
+                self.get_location_str(
+                    schema_definition.get_root_key(), schema_definition.lexemes
+                ),
+            )
 
-            if (
+        if (
+            fully_qualified_name
+            in self.context.context_instance.fully_qualified_name_to_class
+        ):
+            # we've already created the class, so nothing to do here
+            return self.context.context_instance.fully_qualified_name_to_class[
                 fully_qualified_name
-                in context.context_instance.fully_qualified_name_to_class
-            ):
-                # we've already created the class, so nothing to do here
-                return context.context_instance.fully_qualified_name_to_class[
-                    fully_qualified_name
+            ]
+
+        instance_class = None
```

Comment:
```suggestion
```
Really since this is immediately set right after as a return from the `create_instance_class` method, no point initiating it as `None`. And it isn't actioned for anything other than assignment to return within the `create_instance_class` method, so it shouldn't really need to be passed in to get back out either.